integrations/Podsights/browser.js

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable class-methods-use-this */
+import get from "get-value";
+import sha256 from "crypto-js/sha256";
+import { NAME } from "./constants";
+import ScriptLoader from "../ScriptLoader";
+import logger from "../../utils/logUtil";
+import {
+  getHashFromArrayWithDuplicate,
+  removeUndefinedAndNullValues,
+} from "../utils/commonUtils";
+
+class Podsights {
+  constructor(config, analytics) {
+    this.pixelId = config.pixelId;
+    this.eventsToPodsightsEvents = config.eventsToPodsightsEvents;
+    this.enableAliasCall = config.enableAliasCall;
+    this.name = NAME;
+    if (analytics.logLevel) logger.setLogLevel(analytics.logLevel);
+  }
+
+  init() {
+    logger.debug("===In init Podsights===");
+    window.pdst =
+      window.pdst ||
+      function () {
+        (window.pdst.q = window.pdst.q || []).push(arguments);
+      };
+    ScriptLoader("pdst-capture", "https://cdn.pdst.fm/ping.min.js");
+    window.pdst("conf", { key: `${this.pixelId}` });
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded Podsights===");
+    return !!(window.pdst && typeof window.pdst === "function");
+  }
+
+  isReady() {
+    logger.debug("===In isReady Podsights===");
+    return !!(window.pdst && typeof window.pdst === "function");
+  }
+
+  /**
+   * Track - tracks an event for an user
+   * @param {Track} track
+   */
+  track(rudderElement) {
+    logger.debug("===In Podsights Track===");
+    const { message } = rudderElement;
+    const { event, properties } = message;
+    if (!event) {
+      logger.error("[Podsights]: event name from track call is missing.");
+      return;
+    }
+
+    const eventsMapping = getHashFromArrayWithDuplicate(
+      this.eventsToPodsightsEvents,
+      "from",
+      "to",
+      false
+    );
+    const trimmedEvent = event.trim();
+    const events = eventsMapping[trimmedEvent] || [];
+    if (events.length === 0) {
+      logger.error(`===No Podsights Pixel mapped event found. Aborting!===`);
+      return;
+    }
+    const externalId =
+      get(message, "userId") ||
+      get(message, "context.traits.userId") ||
+      get(message, "context.traits.id") ||
+      get(message, "anonymousId");
+
+    events.forEach((podsightEvent) => {
+      if (podsightEvent === "lead") {
+        window.pdst(podsightEvent, { category: trimmedEvent, ...properties });
+      } else {
+        window.pdst(podsightEvent, properties);
```

Comment:
we are sending free flowing field from properties, but according to their documentation [here](https://podsights.com/docs/), they have some specific format, why we are not maintaining that?

Reply:
Added mapping, earlier I thought, the user, will pass all fields from properties. 
And also, should we support other fields apart from mapping?

Reply:
> And also, should we support other fields apart from mapping
we need to confirm this from support team, if fields other than what mentioned in the doc is recorded at all. If yes, we can send.

Reply:
As in the network call, I do not see any other attributes which they accept, and also in the dashboard, some limited fields are visible that we already mapped. So should we ask them? @shrouti1507

Reply:
you mean, the attributes other than whatever they mentioned in the docs?

Reply:
Yes